R/class-spec.R

```diff
@@ -120,12 +140,16 @@ class_desc <- function(x) {
 
 # Vector of class names; used in method introspection
 class_dispatch <- function(x) {
+  if (identical(x, R7_object)) return(c("R7_object", "ANY"))
+
   switch(class_type(x),
-    NULL = NULL,
-    S4 = S4_strip_union(methods::extends(x)),
+    NULL = c("NULL", "ANY"),
+    missing = "MISSING",
+    any = "ANY",
+    S4 = c(S4_strip_union(methods::extends(x)), "ANY"),
     R7 = c(x@name, class_dispatch(x@parent)),
-    R7_base = c("R7_object", x$class),
-    R7_S3 = c("R7_object", x$class),
+    R7_base = c("R7_object", x$class, "ANY"),
+    R7_S3 = c("R7_object", x$class, "ANY"),
```

Comment:
Sort of a strange sandwiching of the base/S3 classes, but I guess this makes sense for dispatch